### PR TITLE
Doc updated README

### LINKED ISSUE #28: Complete the README.md
Say how we use Gradle
How we build the project ¨ -> gradlew build
Say something on ngrok and how to set up the server

### PR DESCRIPTION
**Pull request by myhelmisaari**
Resolves #28 
Adds information about:
-  testing locally using gradle and building locally, using IntelliJ or the terminal
- ngrok: how to set it up and add it to a webhook
- more details on how the server is working

### CODE REVIEW

**Comment by hmellin97** [comment]
I think it looks good.

Small thing under _testing and building locally_:
~~This~~ For windows machines you can use `gradlew test/build`.

File: README.md [hunk 1]
```diff
@@ -1,29 +1,56 @@
...
+This implementation of a CI-server is made with Java SDK 11 and IntelliJ; therefore make sure you have these installed. The project that uses the CI-server needs to have `Gradle` installed, due to it being used by the CI-server. There must exist a `build.gradle` file in the project containing the dependencies needed for the project to build and run.
 
-## Testing
-Tests are written using the `JUnit` library. 
+### Branches
+There are two branches: `master` and `assessment`. The master branch contains the CI-server. The assessment branch is simulating a project with a main class and a simple test for it. The purpose of the assessment branch is to use it to test the functionality of the CI-server. 
+
+## Testing and building locally
+Tests for the CI-server are written using the `JUnit` library. If you want to run the tests locally, there are two ways.
+* **Using the terminal:** first navigate to the the project folder, then use the command `./gradlew test`. To build the project locally use the command `./gradlew build`. For windows machines you can use `gradlew test/build`.
```
Comment (Mathiponds) [inline README.md]: The build command also run the tests. Maybe we can differantiate both like : To only run the test -> `gradle test` but to `gradle build` build and run the tests

File: README.md [hunk 2]
```diff
@@ -1,29 +1,56 @@
...
+* **Using IntelliJ:** Open the project in IntelliJ. Right-click on the src-folder and choose `Run 'Tests in DD2480-Big-Brain-CI.test'`. To build the project right-click on the src-folder and choose  `Build module DD2480-Big-Brain-CI`.
+If the build was successful there will be a new directory in the project named 'Build' that contains the generated classes. If it does not succeed the directory will not contain any generated classes.
```
Comment (Mathiponds) [inline README.md]: `build` instead of `Build`

File: README.md [hunk 3]
```diff
@@ -1,29 +1,56 @@
...
-1. The CI-server will do its part and notify about the results
+1. Commit and push the changes to `GitHub`
+1. The CI-server will do its part and notify about the results of the build and tests using Slack
 
-## What the server is doing
-1. Clone the project using `JGit`, this happens when webhook sees a change (such as a push) in the repository
-1. Compile and run the test using `Gradle` and the Gradle tooling API `GradleConnector` (can be done using the command `gradlew build` in the terminal)
-1. Notify the user using the `XML-file` generated by `Gradle`
+## When the server is activated by the Webhook
+1. Clones the project using `JGit`and saves it to a directory named 'assessment'.
```
Comment (Mathiponds) [inline README.md]: It doesn't use `JGit` anymore. I'm only sending command line from Java
It saves to a directory named `assessmentDir`

File: README.md [hunk 4]
```diff
@@ -1,29 +1,56 @@
...
+1. Builds the cloned project and runs the tests defined in the project. This is done by executing the commands `./gradlew build` and `./gradlew test`.
```
Comment (Mathiponds) [inline README.md]: It executes only the command `gradle build` and nothing else

File: README.md [hunk 5]
```diff
@@ -1,29 +1,56 @@
 # DD2480-CI
-CI is the second assignment in the course DD2480 Software Engineering Fundamentals. The goal of the assignment is to implement a continuous integration server that first clones a repository on GitHub when there are any changes on it, builds the project and notifies the user. 
+CI is the second assignment in the course DD2480 Software Engineering Fundamentals. The goal of the assignment is to implement a continuous integration server that first clones a repository on GitHub when there are any changes pushed on it, builds the project, runs the tests in the project and notifies the user of the results.
```
Comment (axelkennedal) [inline README.md]:
```suggestion
CI is the second assignment in the course DD2480 Software Engineering Fundamentals. The goal of the assignment is to implement a continuous integration server which first clones a repository on GitHub when there are any changes pushed to it, builds the project, runs the tests in the project and notifies the user of the results. 
```

File: README.md [hunk 6]
```diff
@@ -1,29 +1,56 @@
...
+CI is the second assignment in the course DD2480 Software Engineering Fundamentals. The goal of the assignment is to implement a continuous integration server that first clones a repository on GitHub when there are any changes pushed on it, builds the project, runs the tests in the project and notifies the user of the results. 
 
 ## Getting Started
-This implementation of a CI-server is made with Java SDK 11 and IntelliJ; therefore make sure you have these installed. The project that uses the CI-server needs to have `Gradle` installed, due to it being used by the CI-server. There must exist a `build.gradle` file in the project. This file must contain the dependencies needed for the project to run.
+This implementation of a CI-server is made with Java SDK 11 and IntelliJ; therefore make sure you have these installed. The project that uses the CI-server needs to have `Gradle` installed, due to it being used by the CI-server. There must exist a `build.gradle` file in the project containing the dependencies needed for the project to build and run.
```
Comment (axelkennedal) [inline README.md]:
```suggestion
This implementation of a CI-server is made with Java SDK 11 and IntelliJ; therefore make sure you have these installed. The project that uses the CI-server needs to have `Gradle` installed. For the project to build and run, there must exist a `build.gradle` file in the project, containing the dependencies needed.
```

File: README.md [hunk 7]
```diff
@@ -1,29 +1,56 @@
...
 
-## Testing
-Tests are written using the `JUnit` library. 
+### Branches
+There are two branches: `master` and `assessment`. The master branch contains the CI-server. The assessment branch is simulating a project with a main class and a simple test for it. The purpose of the assessment branch is to use it to test the functionality of the CI-server.
```
Comment (axelkennedal) [inline README.md]:
```suggestion
There are two main branches: `master` and `assessment`. The master branch contains the CI-server. The assessment branch is simulating a project with a main class and a simple test for it. The purpose of the assessment branch is to use it to test the functionality of the CI-server. 
```

File: README.md [hunk 8]
```diff
@@ -1,29 +1,56 @@
...
+This implementation of a CI-server is made with Java SDK 11 and IntelliJ; therefore make sure you have these installed. The project that uses the CI-server needs to have `Gradle` installed, due to it being used by the CI-server. There must exist a `build.gradle` file in the project containing the dependencies needed for the project to build and run.
 
-## Testing
-Tests are written using the `JUnit` library. 
+### Branches
+There are two branches: `master` and `assessment`. The master branch contains the CI-server. The assessment branch is simulating a project with a main class and a simple test for it. The purpose of the assessment branch is to use it to test the functionality of the CI-server. 
+
+## Testing and building locally
+Tests for the CI-server are written using the `JUnit` library. If you want to run the tests locally, there are two ways.
+* **Using the terminal:** first navigate to the the project folder, then use the command `./gradlew build`. This command builds and tests the project. If you only want to test and not build, use the command `./gradlew build`. For windows machines you can use `gradlew test/build`.
```
Comment (axelkennedal) [inline README.md]:
```suggestion
* **Using the terminal:** first navigate to the the project folder, then use the command `./gradlew build`. This command builds and tests the project. If you only want to test and not build, use the command `./gradlew test`. For windows machines you can use `gradlew test/build`.
```

File: README.md [hunk 9]
```diff
@@ -1,29 +1,56 @@
...
+All you have to do is open a new terminal window and use the command `./ngrok http 8083`. The ngrok service provides a forwarding URL that works for 8 hours.
+
+After this you have to set up a Webhook. This can be done on the Github repository online.
+1. Go to `Settings >> Webhooks`, click on `Add webhook`.
+1. Paste the forwarding URL that `ngrok` provides (for example `https://7al9b010.ngrok.io`) in the payload field and proceed to click `Add webhook`. Nothing else is required than the URL.
+
+Now the CI-server should be ready to be used. 
+
+### Using the CI-server
+To test the CI-server there is a specific branch prepared. Follow the following steps in the terminal to proceed. These steps assumes you have already cloned the repository.
```
Comment (axelkennedal) [inline README.md]:
```suggestion
To test the CI-server there is a specific branch prepared. Follow the steps below in the terminal to proceed. These steps assumes you have already cloned the repository.
```

**Review by axelkennedal — CHANGES_REQUESTED** [review]
What an excellent README! I wish we had this good documentation at my work 😄

**Review by axelkennedal — APPROVED** [review]

**Pull request merged**